### PR TITLE
migrated project structure to DDD

### PR DESCRIPTION
**Pull request by OlehSarapuk**
i am more than sure that this is not the right way of implementind DDD
but it is the best i can come up with

### CODE REVIEW

**Comment by OlehSarapuk** [comment]
Moved Movie, CinemaHall and MovieSession to DDD fully
And broke down all domains to moduls

File: movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/infrastructure/repository/MovieRepositoryImpl.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    override fun findById(id: String): Mono<Movie> {
+        val query = Query()
+            .addCriteria(Criteria.where("_id").`is`(id))
+        return mongoTemplate.findOne(query, MovieEntity::class.java)
+            .map { mapToDomain(it) }
+    }
+
+    private fun mapToDomain(entity: MovieEntity): Movie {
```
Comment (ajax-shulepa-a) [inline movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/infrastructure/repository/MovieRepositoryImpl.kt]: [nit] good candidate for extension function

File: movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/infrastructure/repository/MovieRepositoryImpl.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package com.example.ajaxonboardingproject.infrastructure.repository
```
Comment (ajax-shulepa-a) [inline movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/infrastructure/repository/MovieRepositoryImpl.kt]: com.example.ajaxonboardingproject.infrastructure.database.repository

File: movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/infrastructure/model/MovieEntity.kt [hunk 3]
```diff
@@ -1,10 +1,10 @@
-package com.example.ajaxonboardingproject.model
+package com.example.ajaxonboardingproject.infrastructure.model
```
Comment (ajax-shulepa-a) [inline movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/infrastructure/model/MovieEntity.kt]: com.example.ajaxonboardingproject.infrastructure.database.model

File: movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/domain/Movie.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.example.ajaxonboardingproject.domain
+
+data class Movie(
+    val title: String,
+    val description: String,
+) {
+    lateinit var id: String
```
Comment (ajax-shulepa-a) [inline movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/domain/Movie.kt]: here you can make id nullable and move to constructor

File: movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/application/grpc/MovieGrpcService.kt [hunk 5]
```diff
@@ -1,10 +1,10 @@
-package com.example.ajaxonboardingproject.grpc
+package com.example.ajaxonboardingproject.application.grpc
```
Comment (ajax-shulepa-a) [inline movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/application/grpc/MovieGrpcService.kt]: grpc is also infrastructure

File: movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/application/repository/MovieRepository.kt [hunk 6]
```diff
@@ -1,6 +1,6 @@
-package com.example.ajaxonboardingproject.repository
+package com.example.ajaxonboardingproject.application.repository
```
Comment (ajax-shulepa-a) [inline movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/application/repository/MovieRepository.kt]: if you want to access from application something that should be in infrastructure - you can provide OutPort interface which will be implemented in infrastructure.

Actually you did this, but let's name it MovieRepositoryOutPort to understand it's nature. and in MovieRepositoryImpl you can omit impl suffix

File: movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/application/service/MovieService.kt [hunk 7]
```diff
@@ -1,6 +1,6 @@
-package com.example.ajaxonboardingproject.service
+package com.example.ajaxonboardingproject.application.service
```
Comment (ajax-shulepa-a) [inline movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/application/service/MovieService.kt:1]: Lets call this class MovieInPort as it's input port to application level from infra level. and again, MovieServiceImpl can become just MovieService then

**Review by ajax-shulepa-a — CHANGES_REQUESTED** [review]

File: user-subdomain/src/test/kotlin/com/example/ajaxonboardingproject/UserRepositoryTests.kt [hunk 8]
```diff
@@ -91,10 +91,13 @@ class UserRepositoryTests {
...
-            Ticket(MovieSession(
+            Ticket(
+                MovieSession(
                 Movie("movie", "awesome"),
-                CinemaHall(100, "awesome"),
-                LocalDateTime.of(2020, 3, 12, 10 , 5))))
+                    CinemaHall(100, "awesome"),
+                LocalDateTime.of(2020, 3, 12, 10 , 5))
+            )
+            )
```
Comment (ajax-gnatiuk-s) [inline user-subdomain/src/test/kotlin/com/example/ajaxonboardingproject/UserRepositoryTests.kt:102]: ugly formatting :)

File: movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/infrastructure/repository/MovieRepositoryImpl.kt (hunk 1 above)
Comment (OlehSarapuk) [reply movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/infrastructure/repository/MovieRepositoryImpl.kt]: Thanks for advice)
Done! ✅

File: movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/infrastructure/repository/MovieRepositoryImpl.kt (hunk 2 above)
Comment (OlehSarapuk) [reply movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/infrastructure/repository/MovieRepositoryImpl.kt]: Done! ✅

File: movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/infrastructure/model/MovieEntity.kt (hunk 3 above)
Comment (OlehSarapuk) [reply movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/infrastructure/model/MovieEntity.kt]: Done! ✅

File: movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/domain/Movie.kt (hunk 4 above)
Comment (OlehSarapuk) [reply movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/domain/Movie.kt]: Done! ✅

File: movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/application/grpc/MovieGrpcService.kt (hunk 5 above)
Comment (OlehSarapuk) [reply movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/application/grpc/MovieGrpcService.kt]: Done! ✅

File: movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/application/repository/MovieRepository.kt (hunk 6 above)
Comment (OlehSarapuk) [reply movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/application/repository/MovieRepository.kt]: Done! ✅

File: movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/application/service/MovieService.kt (hunk 7 above)
Comment (OlehSarapuk) [reply movie-subdomain/src/main/kotlin/com/example/ajaxonboardingproject/application/service/MovieService.kt:1]: Done! ✅

File: user-subdomain/src/test/kotlin/com/example/ajaxonboardingproject/UserRepositoryTests.kt (hunk 8 above)
Comment (OlehSarapuk) [reply user-subdomain/src/test/kotlin/com/example/ajaxonboardingproject/UserRepositoryTests.kt:102]: my bad, refactored ✅

**Review by ajax-shulepa-a — APPROVED** [review]

**Review by Wectro20 — APPROVED** [review]

**Pull request merged**